tests/frontier/blockchain_st_test_helpers.py

```diff
@@ -76,6 +76,13 @@ def run_blockchain_st_test(
 
 def add_blocks_to_chain(chain: BlockChain, test_data: Dict[str, Any]) -> None:
     for idx, block in enumerate(test_data["blocks"]):
+        if idx == 0:
```

Comment:
These assert statements basically check that the test data is correct. This doesn't ensure that the `spec.py` actually validates the parent hash.
What we should be doing is writing a unit test that provides an invalid parent hash and checks that test reverts with a validation error.